src/core/event/poll_group.cpp

```diff
@@ -182,7 +182,15 @@ void poll_group::close_socket(sockinfo_tcp *si, bool force /*=false*/)
 
     bool closed = si->prepare_to_close(force);
     if (closed) {
+        /*
+         * Current implementation forces TCP reset, so the socket is expected to be closable.
+         * Do a polling iteration to increase the chance that all the relevant WQEs are completed
+         * and XLIO emitted all the TX completion before the XLIO_SOCKET_EVENT_TERMINATED event.
+         *
+         * TODO Implement more reliable mechanism of deferred socket destruction if there are
+         * not completed TX operations.
+         */
+        poll();
```

Comment:
AFAIU it will not affect time-wait sockets. So while a socket in time-wait it's TX completions can be triggered?

Reply:
Current implementation doesn't support time wait sockets. Instead, XLIO always forces TCP reset for the xlio_socket_t objetcs. On TCP reset, all the queues (unsent, unacked) are cleared immediately and socket can be destroyed without a delay.

In some future implementation, time-wait state or async TCP FIN handshake will be transparent for users, because API allows TX completions after xlio_socket_destroy() until async event TERMINATED arrives.

Also we cannot wait until socket destructor for TIME WAIT sockets, because it can take 1 minute to generate TERMINATED event. So we will need either speedup TIME WAIT destruction according to RFCs or generate TERMINATED event when socket transits to the TIME WAIT state. However, with the latter solution we will get race for RFC reuse issue.